projects/wave-core-new/src/lib/datasets/dataset/dataset.component.ts

```diff
@@ -26,40 +27,78 @@ export class DataSetComponent implements OnInit {
-                            fillRGBA: this.randomColorService.getRandomColorRgba(),
-                            radius: 10,
-                            clustered: false,
-                        }),
-                        isLegendVisible: false,
-                        isVisible: true,
-                    }));
-                }
-            })
-        ).subscribe();
+                        );
+                    } else {
+                        const result_descriptor = this.dataset.result_descriptor as VectorResultDescriptor;
+
+                        switch (result_descriptor.data_type) {
+                            case VectorDataTypes.MultiPoint:
+                                return this.projectService.addLayer(
+                                    new VectorLayer({
+                                        workflowId,
+                                        name: this.dataset.name,
+                                        symbology: PointSymbology.createSymbology({
+                                            fillRGBA: this.randomColorService.getRandomColorRgba(),
+                                            radius: 10,
+                                            clustered: false,
+                                        }),
+                                        isLegendVisible: false,
+                                        isVisible: true,
+                                    })
+                                );
+
+                            case VectorDataTypes.MultiLineString:
+                                return this.projectService.addLayer(
+                                    new VectorLayer({
+                                        workflowId,
+                                        name: this.dataset.name,
+                                        symbology: LineSymbology.createSymbology({
+                                            strokeRGBA: this.randomColorService.getRandomColorRgba(),
+                                        }),
+                                        isLegendVisible: false,
+                                        isVisible: true,
+                                    })
+                                );
+
+                            case VectorDataTypes.MultiPolygon:
+                                return this.projectService.addLayer(
+                                    new VectorLayer({
+                                        workflowId,
+                                        name: this.dataset.name,
+                                        symbology: VectorSymbology.createSymbology({
+                                            fillRGBA: this.randomColorService.getRandomColorRgba(),
+                                        }),
+                                        isLegendVisible: false,
+                                        isVisible: true,
+                                    })
+                                );
+                        }
```

Comment:
As far as I see the switch-case only affects the symbology, so you should create a symbology based on that and then just create the layer - the rest seems identical.

Reply:
eb7eca6f4cb59e9015cc5931f404c1509f2230fa